### DISCUSSION TITLE
How to validate an array(not registered/without useFieldArray) when using YupResolver

### DISCUSSION THREAD

**Discussion by Moshappp** [react-hook-form/react-hook-form#10007] [Q&A]
I am using React-Hook-Form with yupResolver but I have an array called `studentList` defined in Yup schema that should pass the validation if the array contains at least one `student`. I am not using `FieldArray` as I don't want to render/show the array to the `UI`, the insertion into the array is made through other actions(no directly through UI).

Here is a code snippet
```
const schema = yup
    .object()
    .shape({
      bio: yup
        .string()
        .required("Bio is required"),
      studentList: yup.array().min(1, "This is required"),
      servicesList: yup.array().of(
        yup.object().shape({
          value: yup.string().required("This is required"),
        })
      ),
    })
    .required();
  const {
    register,
    trigger,
    control,
    getValues,
    formState: { errors, isValid },
  } = useForm({
    defaultValues: {
      bio,
      studentList,
      servicesList,
    },
    mode: "onTouched",
    resolver: yupResolver(schema),
  });
```
  
  The array is only get validated based on the first render, for example if the array is empty it is forever going to be invalid even if some data is inserted, and on other hand if the array had at least one item it is going to pass validation check even if the item is deleted.
  
  What I am asking is how to validate this array along with other properties that are registered with React-Hook-Form when submitting.
  
  I tried yup.validate({studentList}) and it didn't work.
  
  Thanks in advance

**Comment by Moshyfawn** [comment]
I won't say I fully understand your issue, but I suppose [the `trigger` form method](https://react-hook-form.com/api/useform/trigger) is what you're looking for?

**Reply by Moshappp** [reply]
Thanks for the answer, the trigger is used for registered fields, while my array is not registered in first place with react-hook-form moreover I don't want to register it as it is not displayed in UI
